### PR TITLE
refactor(ingest/biz-glossary): simplify business glossary source

### PR DESCRIPTION
**Pull request by hsheth2**
- Add status aspects
- Avoid global variables
- Materialize urns up front

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py [hunk 1]
```diff
@@ -1165,6 +1165,9 @@ def __init__(self, looker_api: LookerAPI):
         self.looker_api_wrapper = looker_api
 
     def get_by_id(self, id_: str) -> Optional[LookerUser]:
+        if not id_:
+            return None
+
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:1170]: Is this intentional? Seems like a good change though

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py [hunk 2]
```diff
@@ -332,10 +333,10 @@ def get_mces_from_term(
             models.GlossaryTermKeyClass,
             models.BrowsePathsClass,
         ]
-    ] = []
+    ] = [valid_status]
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py:336]: Would prefer this as `VALID_STATUS` to make it clear it's a global. Also, not familiar with this source at all, I assume we can't use the `auto_status_aspect` helper?

File: metadata-ingestion/src/datahub/ingestion/source/confluent_schema_registry.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 from typing import Any, List, Optional, Set, Tuple
 
 import confluent_kafka
+import confluent_kafka.schema_registry.schema_registry_client
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/confluent_schema_registry.py:8]: Would prefer to do a `from ... import` like we do for `RegisteredSchema`, `Schema`, `SchemaReference`

**Review by asikowitz — COMMENTED** [review]
I can give a deeper review if necessary but I really have no clue what I'm reading. Someone more familiar with this source should probably review instead

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:1170]: nothing added to commit but untracked files present (use "git add" to track)
 datahub/datahub-integrations-service on  no-slacking-hack [$?] is 📦 v0.1.0 via 🐍 v3.10.8 (acryl-datahub-integrations-py3.10)  ❯
gl
remote: Enumeratin

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_common.py:1170]: yup meant to put in a separate PR but it's fine here too

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py:336]: you're right - switching to use the auto_status_aspect instead

File: metadata-ingestion/src/datahub/ingestion/source/confluent_schema_registry.py (hunk 3 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/confluent_schema_registry.py]: updated

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py [hunk 4]
```diff
@@ -472,28 +495,36 @@ def create(cls, config_dict, ctx):
...
         self,
-    ) -> Iterable[Union[MetadataWorkUnit, UsageStatsWorkUnit]]:
+    ) -> Iterable[MetadataWorkUnit]:
         glossary_config = self.load_glossary_config(self.config.file)
-        populate_path_vs_id(glossary_config)
-        for event in get_mces(
-            glossary_config, ingestion_config=self.config, ctx=self.ctx
+
+        materialize_all_node_urns(glossary_config, self.config.enable_auto_id)
+        path_vs_id = populate_path_vs_id(glossary_config)
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py:520]: Can we do these two together? The recursion looks the same and it seems like after the urns are calculated we can just alter `path_vs_id` after we define the urn

**Review by asikowitz — APPROVED** [review]
One suggestion but LGTM

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py (hunk 4 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py:520]: I use them separately here https://github.com/acryldata/business-glossary-sync-action/blob/main/glossary-sync.py#L473, which is why I structured it this way

**Pull request merged**